File: src/mem/BaseSpm.py
# Author: Youngbin Kim

from m5.params import *
from SimpleMemory import *

class BaseSpm(SimpleMemory):
    type = 'BaseSpm'
    cxx_header = "mem/base_spm.hh"

    # each core can access only its own spm
    in_addr_map = False

File: src/mem/base_spm.cc
/*
 * Author: Youngbin Kim
 */

#include "mem/base_spm.hh"

BaseSpm::BaseSpm(const BaseSpmParams* p) :
    SimpleMemory(p) 
{
    //
}

BaseSpm*
BaseSpmParams::create() {
    return new BaseSpm(this);
}
